src/knapsack/BaseKnapsackModel.py

```diff
@@ -112,27 +101,4 @@ def run(self, container, inputDict):
       @ In, container, object, self-like object where all the variables can be stored
       @ In, inputDict, dict, dictionary of inputs from RAVEN
     """
```

Comment:
You can make the run class as abstractmethod

Reply:
added